sysl2/sysl/examples/grammars/go/go.gen.sysl

```diff
@@ -26,16 +32,19 @@ CodeGenTransform:
   !view GoStructFields(fields <: set of Type) -> set of FieldDecl:
     fields -> (field:
       let isKeyword = IsKeyword(field.key).out
-      let prefix = if isKeyword == true then "_" else ""
-      identifier = prefix + field.key
+      let fieldName = GetGoField(field.key).out
+      identifier = fieldName
       let IsPtr = if field.value.optional == true then "*" else ""
       Type = IsPtr + GoType(field.value).out
+      let fieldTag = '`json:"' + ToLower(fieldName)
+      Tag = if field.value.optional == true then (fieldTag + ',omitempty"`') else (fieldTag + '"`')
```

Comment:
feature request ---- probably beyond scope of your change here:

it would be good to allow user who is running codegen to add or customise field tagging for new kinds of tags. these might be user defined tags with some user defined meaning specific to the application.

for example, in addition to `json:` field tags there is another project that uses field tags to declare validations -- https://github.com/go-validator/validator#usage

more tags here: https://github.com/golang/go/wiki/Well-known-struct-tags

Reply:
Thought of that as well. Will add them in next phase

Reply:
Yes, in transform, you can add anything and it will appear on generated code. This is just an example of generating json tags.

Reply:
okay, thanks for explaining.

Reply:
#203